provider/kubernetes/kubernetes.go

```diff
@@ -401,6 +409,100 @@ func (p *Provider) loadConfig(templateObjects types.Configuration) *types.Config
 	return configuration
 }
 
+func (p *Provider) addGlobalBackend(k8sClient Client, i *extensionsv1beta1.Ingress, templateObjects *types.Configuration) error {
```

Comment:
The method name seems incorrect. To the least, the method also populates the frontend, right?

What is the scope of the method, what's it supposed to do? (Its large size _may_ hint at a smell.)

Reply:
The reason I chose this name is in reference to the `backend` section of the spec, which k8s named `global default backend`. If the name bothers us, we can change it in an upcoming refactor :)

Reply:
Ah okay, missed that -- makes sense to me now, so feel free to keep it. :)